dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/DLCOracle.scala

```diff
@@ -193,27 +315,82 @@ case class DLCOracle(private val extPrivateKey: ExtPrivateKeyHardened)(implicit
               s"No event saved with nonce ${nonce.hex} $outcome"))
       }
 
-      eventOutcomeOpt <- eventOutcomeDAO.read((nonce, outcome))
-      eventOutcomeDb <- eventOutcomeOpt match {
-        case Some(value) => Future.successful(value)
-        case None =>
-          Future.failed(new RuntimeException(
-            s"No event outcome saved with nonce and message ${nonce.hex} $outcome"))
+      outcomeBytes = eventDb.eventDescriptorTLV match {
```

Comment:
note to self, review this on round 2 of code review. It seems like we can get rid of the `.asInstanceOf[]` hopefully.

Reply:
We can't match on our type `T` :( need scala 3

Reply:
Found a better way around this